### PR TITLE
Five Lessons Learned blog post

### PR DESCRIPTION
**Pull request by phereford**

### CODE REVIEW

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+layout: post
+title:  "Five Lessons Learned During My First Year as a Technical Director"
+date:   2017-05-01 12:00:00 -0400
+categories: culture management leadership
+author: Patrick Hereford
+---
+
+Making the leap from an Engineer to a Director can be daunting. There are so
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: lowercase engineer and director

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+many other responsibilities that get placed on you as a director that it can be
+overwhelming. The hardest shift, I feel, is the switch from spending most of your
+time and energy coding to solving other types of problems. Most of
+that time shifts towards higher priority responsibilities like: budgeting,
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: no colon

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+planning, product strategy, mentoring, recruitment, and so much more.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: do we have a style for oxford commas? I haven't been using them, but standard journalism style omits them.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+Even though coding is still enjoyable for those making this switch, it can be
+rather far down the list of largest impact a Director can provide to their
+organization. But like engineering, one should have the mindset of seeking
+challenges and continually find learning oppotunities within their new role.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: finding, match `seeking`

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+There will never be a shortage of learning opportunities at this level,
+or so it feels.
+
+With that, here are the top five things I have learned in my time as a Director:
+  
+## Continuous Learning Is a Fantastic Investment
+Continuous learning is relatively new as an HR bonus but its basic premise has
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: What do you mean by "HR Bonus"? Maybe some context here as to what that means to an organization that may not have that in place.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+been around for a while. Here are some things we do that help with continuous
+learning:
+- Office hours for experimentation
+- Conference/workshop budgets for exposure to new ideas and new tools
+- Code review
+- Pair programming
+- Mentoring
+- Technical book a month club (not required reading, but we buy a technical
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: book-a-month

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+book every month)
+
+Thus far the response to the above has led to highly engaged and highly
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: ~~Thus far~~

highly engaged and ~~highly~~ motivated (redundant highly)

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+motivated individuals that continue to apply new tools to new situations and
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: individuals ~~that~~ who

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+are continually creating more sustainable and maintainable software. [Here]()
+is a blog post by one of our engineers speaking about Continuous Learning.
+
+## Breaking Down Knowledge Silos
+As teams grow and scale, knowledge silos tend to form around services,
+codebases, and processes. Certain engineers that build/maintain codebases
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: engineers who ~~that~~

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+end up with all of the knowledge around how that piece of infrastructure
+interacts in the ecosystem. It's important to disseminate this information as
+succinctly as possible. Other than creating documentation in the codebase, we
+have found that using [discourse](https://www.discourse.org/)
+and documenting the following has helped immensely:
+- Business Processes
+- Code Paths
+- Post mortems
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: Post M~~m~~ortems

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+
+Each one of those deserves its own blog post, but suffice to say it helps
+onboarding new engineers to understand how each app works in the ecosystem and
+what it's role is.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: `its`

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+- Code Paths
+- Post mortems
+
+Each one of those deserves its own blog post, but suffice to say it helps
+onboarding new engineers to understand how each app works in the ecosystem and
+what it's role is.  
+
+## Perceived Failure is an Actual Blessing
+Failure is defined differently from person to person. To some, it's about using
+the wrong software pattern in the wrong circumstance. To others, its deploying
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: it's

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+bad code to production. Whatever the case may be, it's important that teams
+understand its absolutely ok to make mistakes. From experience, it seems that
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: it's

OK is capitalized

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+most engineers learn the most from their missteps.
+
+## 1-1 Meetings are Indispensible
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: 1-on-1

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+I never really understood the importance of the 1-1 while being an engineer. You
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: the 1-on-1 ~~while being~~ as an engineer

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+have to take time out of your day and spend 30 minutes with your boss addressing
+things ranging from emotions to tasks. Now as a director, the 1-1 is one of
+the most important set of meetings you will have on a recurring basis. It gives
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: the most important recurring meetings you will have. tighten it up.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 17]
```diff
@@ -0,0 +1,80 @@
...
+you insight into exactly how best to step up and fill in the gaps that the team
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: ~~exactly~~ how best. breaks up the rhythm of the sentence

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+needs, which changes constantly. Most importantly, it sets your team up to be
+successful as you support their needs.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: support its needs. team is singular.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+  
+## The Most Important Aspect of a Director's Job is...
+Listening. I won't expound much on this as many other highly qualified
+individuals have covered this topic at length. Everyone has pain points while
+working at any company. It is important to make sure that individuals feel heard
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: It is important to make sure ~~that~~ individuals feel heard and feel ~~that they are~~ supported within their organization.

**Comment by phereford** [comment]
updated @kelvinma

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 20]
```diff
@@ -0,0 +1,82 @@
...
+author: Patrick Hereford
+---
+
+Making the leap from an engineer to a director can be daunting. There are so
+many other responsibilities that get placed on you as a director that it can be
+overwhelming. The hardest shift, I feel, is the switch from spending most of your
+time and energy coding to solving other types of problems. Most of
+that time shifts towards higher priority responsibilities like budgeting,
+planning, product strategy, mentoring, recruitment and so much more.
+Even though coding is still enjoyable for those making this switch, it can be
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: line break here

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 21]
```diff
@@ -0,0 +1,82 @@
...
+rather far down the list of largest impact a director can provide to their
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: list of largest impacts (plural)

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 22]
```diff
@@ -0,0 +1,82 @@
...
+organization. But like engineering, one should have the mindset of finding
+challenges and continually find learning opportunities within their new role.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: `continually` embracing(?) challenges and ~~continually find~~ learning opportunities

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 23]
```diff
@@ -0,0 +1,82 @@
...
+There will never be a shortage of learning opportunities at this level,
+or so it feels.
+
+With that, here are the top five things I have learned in my time as a director:
+  
+## Continuous Learning Is a Fantastic Investment
+Investment in employee growth is not a new trend. This investment can be
+implemented in a variety of forms ranging from conference budgets to ingrained
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: forms `,` ranging

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 24]
```diff
@@ -0,0 +1,82 @@
...
+- Code review
+- Pair programming
+- Mentoring
+- Technical book-a-month club (not required reading, but we buy a technical
+book every month)
+
+The response to the above has led to highly engaged and motivated individuals
+who continue to apply new tools to new situations and are continually creating
+more sustainable and maintainable software. [Here](https://www.forwardfinancing.com/blog/takeaways-railsconf-2017/)
+is a blog post by one of our engineers speaking about Continuous Learning.
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: by one of our engineers ~~speaking~~ about Continuous Learning.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 25]
```diff
@@ -0,0 +1,82 @@
...
+
+## Breaking Down Knowledge Silos
+As teams grow and scale, knowledge silos tend to form around services,
+codebases, and processes. Certain engineers who build and maintain codebases
+end up with all of the knowledge around how that piece of infrastructure
+interacts in the ecosystem. It's important to disseminate this information as
+succinctly as possible. Other than creating documentation in the codebase, we
+have found that using [discourse](https://www.discourse.org/)
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: Capitalize Discourse

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 26]
```diff
@@ -0,0 +1,82 @@
...
+Failure is defined differently from person to person. To some, it's about using
+the wrong software pattern in the wrong circumstance. To others, it's deploying
+bad code to production. Whatever the case may be, it's important that teams
+understand it's absolutely OK to make mistakes. From experience, it seems that
+most engineers learn the most from their missteps.
+
+## 1-on-1 Meetings are Indispensible
+I never really understood the importance of the 1-on-1 as an engineer. You
+have to take time out of your day and spend 30 minutes with your boss addressing
+things ranging from emotions to tasks. Now as a director, the 1-1 is the
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: 1-on-1

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 27]
```diff
@@ -0,0 +1,82 @@
...
+most important recurring meeting meetings you will have. It gives
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: recurring ~~meeting~~ meetings

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 28]
```diff
@@ -0,0 +1,82 @@
...
+you insight into how best to step up and fill the gaps that the team may have,
+which changes constantly. Most importantly, it sets your team up to be
+successful as you support its needs.
+  
+## The Most Important Aspect of a Director's Job is...
+Listening. I won't expound much on this as many other highly qualified
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: Line break after `Listening` for emphasis.

File: _posts/2017-05-01-five-lessons-learned.markdown [hunk 29]
```diff
@@ -0,0 +1,82 @@
...
+individuals have covered this topic at length. Everyone has pain points while
+working at any company. It is important to make sure individuals feel heard
+and feel that they are supported within their organization. If any of these
```
Comment (kelvinma) [inline _posts/2017-05-01-five-lessons-learned.markdown]: feel ~~that they are~~ supported

**Comment by phereford** [comment]
@kelvinma updated grammar issues. In checking the other posts, we use `##` in markdown for our subheadings. That's what I am using here so I am surprised they are bigger. Ill look into that a little more on Monday.

**Pull request merged**